go/6_auth-kubernetes.go

```diff
@@ -35,39 +43,37 @@ import (
 //     	bound_service_account_names="my-app" \
 //		bound_service_account_namespaces="default"
 func getSecretWithKubernetesAuth() (string, error) {
-	// If set, the VAULT_ADDR environment variable will be the address that your pod uses to communicate with Vault.
+	// If set, the VAULT_ADDR environment variable will be the address that
+	// your pod uses to communicate with Vault.
 	config := vault.DefaultConfig() // modify for more granular configuration
 
 	client, err := vault.NewClient(config)
 	if err != nil {
 		return "", fmt.Errorf("unable to initialize Vault client: %w", err)
 	}
 
-	// Read the service-account token from the path where the token's Kubernetes Secret is mounted.
-	// By default, Kubernetes will mount this to /var/run/secrets/kubernetes.io/serviceaccount/token
-	// but an administrator may have configured it to be mounted elsewhere.
-	jwt, err := os.ReadFile("path/to/service-account-token")
+	// The service-account token will be read from the path where the token's
+	// Kubernetes Secret is mounted. By default, Kubernetes will mount it to
+	// /var/run/secrets/kubernetes.io/serviceaccount/token, but an administrator
+	// may have configured it to be mounted elsewhere.
+	// In that case, we'll use the option WithServiceAccountTokenPath to look
+	// for the token there.
+	k8sAuth, err := auth.NewKubernetesAuth(
+		"dev-role-k8s",
+		auth.WithServiceAccountTokenPath("path/to/service-account-token"),
+	)
 	if err != nil {
-		return "", fmt.Errorf("unable to read file containing service account token: %w", err)
+		return "", fmt.Errorf("unable to initialize Kubernetes auth method: %w", err)
 	}
 
-	params := map[string]interface{}{
-		"jwt":  string(jwt),
-		"role": "dev-role-k8s", // the name of the role in Vault that was created with this app's Kubernetes service account bound to it
-	}
-
-	// log in to Vault's Kubernetes auth method
-	resp, err := client.Logical().Write("auth/kubernetes/login", params)
+	authInfo, err := client.Auth().Login(context.TODO(), k8sAuth)
 	if err != nil {
-		return "", fmt.Errorf("unable to log in with Kubernetes auth: %w", err)
+		return "", fmt.Errorf("unable to login to Kubernetes auth method: %w", err)
```

Comment:
This is a nit pick, but shouldn't this still be `with` here b/c the method is the means with which you are logging in? haha. My grammar may be completely wrong

Reply:
You're right, I don't actually remember making this change or why I might have, but I'll change it back haha.